controllers/feeds/getFeedsV2.js

```diff
@@ -94,6 +94,71 @@ const isValidActivity = async (item, conditions) => {
   return (item.final_score || 0) >= threshold;
 };
 
+const getActorFromLatestReaction = (latestReaction) => {
+  let postActors = [];
+  if (latestReaction?.upvotes?.length > 0) {
+    for (let j = 0; j < latestReaction?.upvotes.length; j++) {
+      if (!postActors.includes(latestReaction.upvotes[j].user_id)) {
+        postActors.push(latestReaction.upvotes[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.downvotes?.length > 0) {
+    for (let j = 0; j < latestReaction?.downvotes.length; j++) {
+      if (!postActors.includes(latestReaction.downvotes[j].user_id)) {
+        postActors.push(latestReaction.downvotes[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.comments?.length > 0) {
+    for (let j = 0; j < latestReaction?.comments.length; j++) {
+      if (!postActors.includes(latestReaction.comments[j].user_id)) {
+        postActors.push(latestReaction.comments[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.comment?.length > 0) {
+    for (let j = 0; j < latestReaction?.comment.length; j++) {
+      if (!postActors.includes(latestReaction.comment[j].user_id)) {
+        postActors.push(latestReaction.comment[j].user_id);
+      }
+    }
+  }
+  return postActors;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getActorFromLatestReaction` function is iterating over the upvotes, downvotes, comments, and comment arrays separately to find unique user_ids. This can be optimized by using a Set data structure which automatically ensures uniqueness of elements. Also, you can use the `forEach` method instead of the traditional `for` loop for better readability.

```diff
-  let postActors = [];
+  let postActors = new Set();
   if (latestReaction?.upvotes?.length > 0) {
-    for (let j = 0; j < latestReaction?.upvotes.length; j++) {
-      if (!postActors.includes(latestReaction.upvotes[j].user_id)) {
-        postActors.push(latestReaction.upvotes[j].user_id);
-      }
-    }
+    latestReaction.upvotes.forEach(upvote => postActors.add(upvote.user_id));
   }
   // Similar changes for downvotes, comments, and comment
-  return postActors;
+  return Array.from(postActors);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->